packages/serialization/json/test/common/unionOfObjectsAndPrimitives.ts

```diff
@@ -0,0 +1,216 @@
+	return deserializeIntoPetGetResponse;
+}
+/**
+ * The deserialization information for the current model
+ * @returns {Record<string, (node: ParseNode) => void>}
+ */
+export function deserializeIntoPetGetResponse(petGetResponse: Partial<PetGetResponse> | undefined = {}): Record<string, (node: ParseNode) => void> {
+	return {
+		data: (n) => {
+			petGetResponse.data = n.getNumberValue() ?? n.getStringValue() ?? n.getObjectValue<Cat | Dog | number | string>(createPetGetResponse_dataFromDiscriminatorValue);
+		},
+		request_id: (n) => {
+			petGetResponse.request_id = n.getStringValue();
+		},
+	};
+}
+/**
+ * The deserialization information for the current model
+ * @returns {Record<string, (node: ParseNode) => void>}
+ */
+export function deserializeIntoPetGetResponse_data(petGetResponse_data: Partial<Cat | Dog | number | string> | undefined = {}): Record<string, (node: ParseNode) => void> {
+	return {
+		...deserializeIntoCat(petGetResponse_data as Cat),
+		...deserializeIntoDog(petGetResponse_data as Dog),
+	};
+}
+export interface PetGetResponse extends AdditionalDataHolder, Parsable {
+	/**
+	 * Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
+	 */
+	additionalData?: Record<string, unknown>;
+	/**
+	 * The data property
+	 */
+	data?: Cat | Dog | number | string;
+	/**
+	 * The request_id property
+	 */
+	request_id?: string;
+}
+export type PetGetResponse_data = Cat | Dog | number | string;
+/**
+ * Builds and executes requests for operations under /pet
+ */
+export interface PetRequestBuilder extends BaseRequestBuilder<PetRequestBuilder> {
+	/**
+	 * Get pet information
+	 * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
+	 * @returns {Promise<PetGetResponse>}
+	 */
+	get(requestConfiguration?: RequestConfiguration<object> | undefined): Promise<PetGetResponse | undefined>;
+	/**
+	 * Get pet information
+	 * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
+	 * @returns {RequestInformation}
+	 */
+	toGetRequestInformation(requestConfiguration?: RequestConfiguration<object> | undefined): RequestInformation;
+}
+/**
+ * Serializes information the current object
+ * @param writer Serialization writer to use to serialize this model
+ */
+export function serializePetGetResponse(writer: SerializationWriter, petGetResponse: Partial<PetGetResponse> | undefined = {}): void {
+	switch (typeof petGetResponse.data) {
+		case "number":
+			writer.writeNumberValue("data", petGetResponse.data);
+			break;
+		case "string":
+			writer.writeStringValue("data", petGetResponse.data);
+			break;
+		default:
+			writer.writeObjectValue<Cat | Dog | number | string>("data", petGetResponse.data, serializePetGetResponse_data);
```

Comment:
shouldn't the generic type parameter here contain only the object types?

Reply:
This is resolved in the latest generation.

Reply:
Can you push an update please? and resolve the conflicts as well?